File: backend/src/main/java/com/elhabhab/backend/dto/response/CouponResponseDTO.java
package com.elhabhab.backend.dto.response;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CouponResponseDTO {

    private Long id;
    private UUID couponId;
    private String code;
    private BigDecimal discountAmount;
    private Boolean percentage;
    private LocalDateTime validFrom;
    private LocalDateTime validUntil;
    private int usageLimit;
    private int usedCount;
    private boolean active;
}

File: frontend-angular/src/app/interface/coupon-response-dto.ts
// coupon-response-dto.ts
export interface CouponResponseDTO {
  id: number;
  couponId: string;
  code: string;
  discountAmount: number;
  percentage: boolean;
  validFrom: string;
  validUntil: string;
  usageLimit: number;
  usedCount: number;
  active: boolean;
}
